lib/constable_web/controllers/announcement_controller.ex

```diff
@@ -27,11 +31,16 @@ defmodule ConstableWeb.AnnouncementController do
   end
 
   def index(conn, params) do
-    index_page = my_announcements(conn) |> Repo.paginate(params)
+    index_page = announcements_of_interest(conn) |> Repo.paginate(params)
 
     conn
-    |> assign(:announcements, index_page.entries)
     |> assign(:show_all, false)
+    |> render_index(index_page)
+  end
+
+  defp render_index(conn, index_page) do
```

Comment:
Decided to extract a lot of the common rendering logic from the `def index/2` functions in here.